src/event/input.ts

```diff
@@ -220,6 +222,35 @@ function calculateNewValue(
     }
   }
 
+  if (
+    isElementType(node, 'input', {type: 'number'} as const) &&
+    inputType === 'changeNumberInput' &&
+    !isDisabled(node) &&
+    !node.readOnly
+  ) {
+    const step = node.step ? Number(node.step) : 1
+
+    const reachedMax = value === node.max
+    if (inputData === 'ArrowUp' && !reachedMax) {
+      const exceedsMax = Number(value) + step > Number(node.max)
```

Comment:
What if `value` is empty? `Number(value)` will be `NaN`

Reply:
in that case I think value should be treated as "zero" (seems like that's what browsers do)